SU2_CFD/src/numerics/CLookUpTable.cpp

```diff
@@ -0,0 +1,593 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../include/numerics/CLookUpTable.hpp"
+
+CLookUpTable::CLookUpTable(string var_file_name_lut, string name_prog, string name_enth) {
+  file_name_lut = var_file_name_lut;
+
+  rank = SU2_MPI::GetRank();
+
+  LoadTableRaw(var_file_name_lut);
+
+  FindTableLimits(name_prog, name_enth);
+
+  if (rank == MASTER_NODE)
+    cout << "Detecting all unique edges and setting edge to triangle connectivity "
+            "..."
+         << endl;
+
+  IdentifyUniqueEdges();
+
+  if (rank == MASTER_NODE) cout << " done." << endl;
+
+  PrintTableInfo();
+
+  if (rank == MASTER_NODE)
+    cout << "Building a trapezoidal map for the (progress variable, enthalpy) "
+            "space ..."
+         << endl;
+
+  trap_map_prog_enth = CTrapezoidalMap(GetData(name_prog), GetData(name_enth), GetEdges(), GetEdgeToTriangle());
+
+  if (rank == MASTER_NODE) cout << " done." << endl;
+
+  if (rank == MASTER_NODE) {
+    cout << "Precomputing interpolation coefficients..." << endl;
+  }
+
+  ComputeInterpCoeffs(name_prog, name_enth);
+
+  if (rank == MASTER_NODE) cout << "LUT fluid model ready for use" << endl;
+}
+
+void CLookUpTable::LoadTableRaw(string var_file_name_lut) {
+  CFileReaderLUT file_reader;
+
+  if (rank == MASTER_NODE) cout << "Loading look-up-table-file " << var_file_name_lut << " ..." << endl;
+
+  file_reader.ReadRawDRG(var_file_name_lut);
+
+  n_points = file_reader.GetNPoints();
+  n_triangles = file_reader.GetNTriangles();
+  n_variables = file_reader.GetNVariables();
+  n_hull_points = file_reader.GetNHullPoints();
+  type_lut = file_reader.GetTypeLUT();
+  version_lut = file_reader.GetVersionLUT();
+  version_reader = file_reader.GetVersionReader();
+
+  names_var = file_reader.GetNamesVar();
+  table_data = file_reader.GetTableData();
+  triangles = file_reader.GetTriangles();
+  hull = file_reader.GetHull();
+
+  if (rank == MASTER_NODE) cout << " done." << endl;
+}
+
+void CLookUpTable::FindTableLimits(string name_prog, string name_enth) {
+  int ixEnth = GetIndexOfVar(name_enth);
+  int ixProg = GetIndexOfVar(name_prog);
+
+  limits_table_enth[0] = *min_element(table_data.at(ixEnth).begin(), table_data.at(ixEnth).end());
+  limits_table_enth[1] = *max_element(table_data.at(ixEnth).begin(), table_data.at(ixEnth).end());
+  limits_table_prog[0] = *min_element(table_data.at(ixProg).begin(), table_data.at(ixProg).end());
+  limits_table_prog[1] = *max_element(table_data.at(ixProg).begin(), table_data.at(ixProg).end());
+}
+
+void CLookUpTable::PrintTableInfo() {
+  if (rank == MASTER_NODE) {
+    cout << setfill(' ');
+    cout << endl;
+    cout << "+------------------------------------------------------------------+" << endl;
+    cout << "|                     Look-Up-Table (LUT) info                     |" << endl;
```

Comment:
\n and endl on the last print, endl includes a flush
```suggestion
    cout << "+------------------------------------------------------------------+\n";
    cout << "|                     Look-Up-Table (LUT) info                     |\n";
```